// graphql-date-scalars/build/scalars/date.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DateScalar = void 0;
const utils_1 = require("../utils");
const type_guards_1 = require("../utils/type-guards");
const types_1 = require("../utils/types");
/**
 * An RFC 3339 compliant date-time scalar.
 *
 * Input:
 *    This scalar takes an RFC 3339 date-time string as input and
 *    parses it to a javascript Date.
 *
 * Output:
 *    This scalar serializes javascript Dates,
 *    to RFC 3339 UTC date-time strings.
 */
const dateScalar = new types_1.GraphQLDateScalarType({
    name: 'Date',
    description: 'A date string, such as 2007-12-03, compliant with the `full-date` ' +
        'format outlined in section 5.6 of the RFC 3339 profile of the ' +
        'ISO 8601 standard for representation of dates and times using ' +
        'the Gregorian calendar.',
    serialize(value) {
        if (!(value instanceof Date)) {
            throw new TypeError('Date cannot represent non-date type');
        }
        if ((0, utils_1.validateJSDate)(value)) {
            return (0, utils_1.serializeDate)(value);
        }
        throw new TypeError('Date cannot represent an invalid Date instance');
    },
    parseValue(value) {
        if (typeof value !== 'string') {
            throw new TypeError(`Date cannot represent non string type ${JSON.stringify(value)}`);
        }
        const trimmedValue = value.split('T')[0];
        if ((0, utils_1.validateDate)(trimmedValue)) {
            return (0, utils_1.parseDate)(trimmedValue);
        }
        throw new TypeError(`Date cannot represent an invalid date-string ${value}.`);
    },
    parseLiteral(ast) {
        if (!(0, type_guards_1.isStringValueNode)(ast)) {
            throw new TypeError(`Date cannot represent non string type ${ast.kind}`);
        }
        const { value } = ast;
        const trimmedValue = value.split('T')[0];
        if ((0, utils_1.validateDate)(trimmedValue)) {
            return (0, utils_1.parseDate)(trimmedValue);
        }
        throw new TypeError(`Date cannot represent an invalid date-string ${String(value)}.`);
    },
});
exports.DateScalar = dateScalar;
